src/components/legend/style.js

```diff
@@ -52,7 +52,9 @@ module.exports = function style(s, gd) {
     .each(styleBoxes)
     .each(stylePies)
     .each(styleLines)
-    .each(stylePoints);
+    .each(stylePoints)
+    .each(styleCandles)
+    .each(styleOHLC);
```

Comment:
:rotating_light:  Non-blocking thinking-out-loud comment alert :rotating_light: 

We should perhaps make these trace module method (e.g. named `_module.legendStyle`) instead of having the legend component handle all trace types by itself.

Reply:
Good idea, that would be way more efficient too - instead of calling every style routine on every trace and quitting if it's not the right one, just something like:

```js
.each(function(d) { d.trace._module.legendStyle(...) })
```